tasks/docker_tasks.py

```diff
@@ -92,6 +92,8 @@ def dockerize_test(ctx, binary, skip_cleanup=False):
             '/proc': {'bind': '/host/proc', 'mode': 'ro'},
             '/sys/fs/cgroup': {'bind': '/host/sys/fs/cgroup', 'mode': 'ro'},
             scratch_volume.name: {'bind': '/tmp/scratch', 'mode': 'rw'},
+            # Forwarding the Docker config to the container not to hit rate limits
+            '/root/.docker': {'bind': '/root/.docker', 'mode': 'rw'},
```

Comment:
```suggestion
            '/root/.docker': {'bind': '/root/.docker', 'mode': 'ro'},
```
I think readonly is enought there

Reply:
Discussed offline:
- Binding just the file made the CI think that `/root/.docker/config.json` was a folder ([see logs](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/705259714)):
```
[Error] fail to pull: exit status 18 WARNING: Error loading config file: /root/.docker/config.json: read /root/.docker/config.json: is a directory
```
- The `ro` mode on the whole `/root/.docker` folder made it [fail in CircleCI](https://app.circleci.com/pipelines/github/DataDog/datadog-agent/204421/workflows/c8e6b632-7da6-4dfa-9a79-24e7982c60db/jobs/1928745) because the tests is trying to write `/root/.docker/buildx` in the container.

Trying to bind just the config.json file with [b28a013](https://github.com/DataDog/datadog-agent/pull/30856/commits/b28a0136e146185566a0dba7446a21039cbb23d6).

Reply:
Tried it in [b28a013](https://github.com/DataDog/datadog-agent/pull/30856/commits/b28a0136e146185566a0dba7446a21039cbb23d6), keeping it as is.